consensus/src/dag/dag_store.rs

```diff
@@ -232,6 +233,29 @@ impl Dag {
             })
     }
 
+    pub fn reachable_from_target_node(
+        &self,
+        from: &Arc<CertifiedNode>,
+        until: Option<Round>,
+        filter: impl Fn(&NodeStatus) -> bool,
+    ) -> impl Iterator<Item = &NodeStatus> {
+        self.reachable(vec![from.digest()], from.metadata().round(), until, filter)
+    }
+
+    pub fn reachable_from_parents(
+        &self,
+        parents: &[NodeMetadata],
+        until: Option<Round>,
+        filter: impl Fn(&NodeStatus) -> bool,
+    ) -> impl Iterator<Item = &NodeStatus> {
+        self.reachable(
+            parents.iter().map(|p| *p.digest()).collect(),
+            parents[0].round(),
+            until,
+            filter,
```

Comment:
Instead of two helper functions, maybe just pass Vec of targets to reachable() and have this code there to create the initial?

Reply:
the filter seems very general, I wonder should we just pass which NodeStatus we want to accept here instead of a general Fn -> bool